File: Evergrande/src/gestorAplicación/ApartaEstudio.java
import gestorAplicaci.Inmueble;

public class ApartaEstudio extends Inmueble{
    private String nombreUnidad;
    private int torre;
    private int numeroApto;
    private int numHabitaciones;
    private int numBanos;

    public ApartaEstudio(){}

    public ApartaEstudio(double precio, String direccion, double area, String nombreUnidad, int torre, int numeroApto, int numHabitaciones, int numBanos){
        super(precio, direccion, area);

        this.nombreUnidad = nombreUnidad;
        this.torre = torre;
        this.numeroApto = numeroApto;
        this.numHabitaciones = numHabitaciones;
        this.numBanos = numBanos;
    }

    public String getNombreUnidad() {
        return nombreUnidad;
    }
    public void setNombreUnidad(String nombreUnidad) {
        this.nombreUnidad = nombreUnidad;
    }
    public int getTorre() {
        return torre;
    }
    public void setTorre(int torre) {
        this.torre = torre;
    }
    public int getNumeroApto() {
        return numeroApto;
    }
    public void setNumeroApto(int numeroApto) {
        this.numeroApto = numeroApto;
    }
    public int getNumHabitaciones() {
        return numHabitaciones;
    }
    public void setNumHabitaciones(int numHabitaciones) {
        this.numHabitaciones = numHabitaciones;
    }
    public int getNumBanos() {
        return numBanos;
    }
    public void setNumBanos(int numBanos) {
        this.numBanos = numBanos;
    }

    
}

File: Python/src/gestorAplicacion/herencia/Apartaestudio.py
'''
Autores: 
Julián Orozco Vanegas
Juan Nicolas Piedrahita Salas
David Escobar Ruiz

Esta clase simula el comportamiento de un inmueble tipo apartaestudio y sus caracteristicas
'''

from .Inmueble import Inmueble


class ApartaEstudio(Inmueble):
    def __init__(self, precio, direccion, area, numPisos, parqueaderoCarros, parquederoMotos, amueblado, tipoContrato, idUnidad, nombreUnidad, torre, numeroApto, numHabitaciones, numBanos):
        super().__init__(precio, direccion, area, numPisos, parqueaderoCarros, parquederoMotos, amueblado, tipoContrato, idUnidad)
        self._nombreUnidad = nombreUnidad
        self._torre = torre
        self._numeroApto = numeroApto
        self._numHabitaciones = numHabitaciones
        self._numBanos = numBanos
    
    #toString
    def __str__(self):
        return "| {:<4d} | {:<13s} | {:<14f} | {:<15s} | {:<6f} | {:<11s} | {:<19s} | {:<17s} | {:<20s} | {:<20s} | {:<7s} | {:<20s} | {:<21d} | {:<14d} |".format(self.getIdInmueble(),
                                                                                                                                                                   type(self).__name__,
                                                                                                                                                                   self.getPrecio(),
                                                                                                                                                                   self.getTipoContrato(),
                                                                                                                                                                   self.getArea(),
                                                                                                                                                                   self.getAmueblado(),
                                                                                                                                                                   self.getParqueaderoCarros(),
                                                                                                                                                                   self.getParqueaderoMotos(),
                                                                                                                                                                   self.getDireccion(),
                                                                                                                                                                   self.getNombreUnidad(),
                                                                                                                                                                   self.getTorre(),
                                                                                                                                                                   self.getNumeroApto(),
                                                                                                                                                                   self.getNumHabitaciones(),
                                                                                                                                                                   self.getNumBanos())
        
        
    #Getters and Setters
    def getNombreUnidad(self):
        return self._nombreUnidad
    
    def setNombreUnidad(self,nombre)        :
        self._nombreUnidad = nombre
        
    def getTorre(self):
        return self._torre
    
    def setTorre(self,torre):
        self._torre = torre
        
    def getNumeroApto(self):
        return self._numeroApto
    
    def setNumeroApto(self,apto):
        self._numeroApto = apto
        
    def getNumHabitaciones(self):
        return self._numHabitaciones
    
    def setNumHabitaciones(self, num):
        self._numHabitaciones = num
        
    def getNumBanos(self):
        return self._numBanos
    
    def setNumBanos(self, num):
        self._numBanos = num
        
    